sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/utils/JsonSerializer.java

```diff
@@ -52,13 +54,29 @@ public JsonSerializer() {
     @Override
     public <T> T deserializeFromBytes(byte[] bytes, Type type) throws IOException {
         try (JsonReader jsonReader = JsonReader.fromBytes(bytes)) {
-            if (type instanceof Class<?> && JsonSerializable.class.isAssignableFrom(TypeUtil.getRawClass(type))) {
+            if (type instanceof ParameterizedType && List.class.isAssignableFrom(TypeUtil.getRawClass(type))) {
```

Comment:
This looks good to handle `List<JsonSerializableType>` but nothing more than that. This is good to get us unblocked but we might want to look into more generalized handling in the future for stuff like `Map<String, JsonSerializableType>` or `List<List<Type>>`.

Reply:
Yeah, from a general serialization perspective, we should handle all those scenarios. However, for client libraries, we don't have models that are of `List<List<Type>> ` type. So, this should handle most of the pageable responses.

Reply:
Created an issue to follow up - https://github.com/Azure/azure-sdk-for-java/issues/44213